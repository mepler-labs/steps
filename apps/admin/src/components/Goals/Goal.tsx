import Panel from 'atoms/Panel';
import React from 'react';
import EditButton from 'atoms/Buttons/EditButton';
import styled from 'styled-components';
import { pink } from 'styles/colors';
import { remCalc } from 'styles/type';
import { Flex } from 'grid-styled';

interface Props {
  text: string;
  onEdit(): void;
  className?: string;
}

const Goal: React.SFC<Props> = ({ text, onEdit, className }) => (
  <Panel className={className}>
    <Flex justifyContent="flex-end">
      <EditButton component="Goal" onClick={onEdit} />
    </Flex>
    <Flex flexDirection="column" alignItems="center">
      <StarIcon className="material-icons">star</StarIcon>
      <StyledText>{text}</StyledText>
    </Flex>
  </Panel>
);

const StarIcon = styled.i`
  color: ${pink};
  font-size: ${remCalc(50)};
`;

const StyledText = styled.div`
  text-align: center;
  font-size: ${remCalc(64)};
`;

export default Goal;
