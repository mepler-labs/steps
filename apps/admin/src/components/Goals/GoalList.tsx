import React from 'react';
import Goal from './Goal';
import styled from 'styled-components';
import { Client } from 'reducers/clients';
import { History, withRouter } from 'react-router-dom';

interface Props {
  className?: string;
  goals: string[];
  client: Client;
  history: History;
}

export const GoalList: React.SFC<Props> = ({ goals, client, history }) => (
  <div>
    {goals.map((goal, i) => (
      <StyledGoal
        text={goal}
        key={i}
        onEdit={() => {
          history.push(`/clients/${client.id}/goals/${i}/edit`);
        }}
      />
    ))}
  </div>
);

export const StyledGoal = styled(Goal)`
  margin-bottom: 2rem;
`;

export default withRouter(GoalList);
