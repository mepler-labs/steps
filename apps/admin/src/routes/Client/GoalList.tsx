import Goal from 'components/Goals/Goal';
import { GoalList } from 'components/Goals/GoalList';
import React from 'react';
import { connect, History } from 'react-redux';

interface Props {
  goals: string[];
}

export class Goals extends React.Component<Props> {
  render() {
    const { goals } = this.props;
    return (
      <GoalList goals={goals}>
        {childProps => <Goal {...childProps} />}
      </GoalList>
    );
  }
}
const mapStateToProps = state => ({
  goals: state.auth.user.goals,
});

export default connect(mapStateToProps)(Goals);
