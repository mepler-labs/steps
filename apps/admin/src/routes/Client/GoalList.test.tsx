import 'jest';
import React from 'react';
import { shallow } from 'enzyme';
import { Goals } from './GoalList';
import Goal from 'components/Goals/Goal';

describe('ToalsList.tsx', () => {
  const goals: string[] = ['first goal', 'second goal'];
  const history = {
    push: jest.fn(),
  };

  it('displays goals', () => {
    const wrapper = shallow(<Goals history={history} goals={goals} />);

    expect(wrapper).toMatchSnapshot();
    const goalsList = wrapper.dive().find(Goal);

    expect(goalsList).toHaveLength(2);
  });

  it('routes to /goals/edit', () => {
    const wrapper = shallow(<Goals history={history} goals={goals} />);
    const goalsList = wrapper.dive().find(Goal);

    goalsList.first().simulate('edit');

    expect(history.push).toHaveBeenCalledWith('/goals/edit');
  });
});
