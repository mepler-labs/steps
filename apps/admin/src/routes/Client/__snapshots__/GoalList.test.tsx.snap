// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToalsList.tsx displays goals 1`] = `
<Component
  goals={
    Array [
      "first goal",
      "second goal",
    ]
  }
/>
`;
